### ISSUE TITLE
Use javax.persistence.Column annotation to map resultSet

### ISSUE THREAD

**Issue by amanteaux** [aaberg/sql2o#204]
It would ease the transition between Hibernate/JPA to sql2o if sql2o were able to read the JPA Column annotation to map the resultSet to the POJO.

I can take care of this feature, but before I do anything, I would like to make sure that:
- this feature does not already exist,
- you agree with the implementation I want to do.

Actually I have found 2 options:
- either this feature is implemented with a **tight integration to sql2o**. This way, the implementation can be done directly in `PojoMetadata`, the JPA annotations can be read alongside the rest of the POJO information, in `initializePropertyInfo()`.
- either this feature is **not integrated by default to sql2o**. This way, the feature should be an implementation of `ResultSetHandlerFactoryBuilder`. However the new `ResultSetHandlerFactoryBuilder` is currently difficult to be attached to the `Sql2o` instance: one has to get a new connection, create a new query and then it is possible to specify the new `ResultSetHandlerFactoryBuilder`. So if this feature is integrated in a new `ResultSetHandlerFactoryBuilder`, it would be better if the `ResultSetHandlerFactoryBuilder` could be set directly to the `Sql2o` instance.

What are your thoughts?

**Comment by aaberg** [comment]
Hi,

This feature does not already exist, but I think it would be a really nice addition to sql2o.

Since it does not require any new dependencies, I think the best solution would be the tight integration. The logic should be to first check for the `@Column` annotation and use that. If there is no `@Column` annotation, it should use existing logic. Do you agree?

**Comment by dimzon** [comment]
public interface ColumnNamingStrategy{ String getName(Field f) ;}
23.11.2015 11:47 пользователь "Lars Aaberg" <notifications@github.com>
написал:

**Comment by amanteaux** [comment]
Yes I agree, the `@Column` annotation should be used first if it exists, else the existing logic should be used.

By the way, I noticed that in `DefaultResultSetHandlerFactory` the `getGetter` method and the getter part in `newResultSetHandler0()` are not used. I would like to remove the corresponding code (maybe in another github issue), but is this code intended to be used in the future ?
I also noticed the same issue with the `Pojo` class.
